### ISSUE TITLE
NodeBots at CampJS III

### ISSUE THREAD

**Issue by AnnaGerber** [campjs/campjs-iii#10]
Continuation of discussion at https://github.com/ajfisher/nbdau/issues/10

I put in a workshop proposal on this topic yesterday, keen to be involved:) 

I can bring my own experimenters kits (upgraded to less dodgy Arduino clones than last time) and we could also use the NBD SimpleBot kits. 

I'm working on adding more mobile robot specific exercises to node-ardx at the moment (locomotion, sensors etc)

**Comment by ajfisher** [comment]
Cool - I'd be happy to help run it with you @AnnaGerber 

So the biggest concern I have is really the timing around getting people up and running versus their "productive" time afterwards. As we've seen at all the NodeBots events those who's environments "just work" tend to get cracking inside 30 minutes but those who don't can end up taking a solid hour or two. With a 90 minute window that may cause challenges.

I know @timoxley had mentioned that the 90 mins was a guide so I just wanted to float the idea of doing a "double" session and see what sort of support there was for that.

Regarding the kit, I have 6 SimpleBot kits available for Melbourne so between us we can muster up 10-12. Again, being conscious of time, I wonder whether we just have them built so people can concentrate on programming them? This was the other big time sink for most people as they'd never done any electronics.

Having the experimenters kits available then for people who are more interested in learning the electronics side would support that approach and be something they could tinker away with.

Thoughts on this approach?

**Comment by nog3** [comment]
I'll be attending too, so I'll put my hand up to help out.

**Comment by AnnaGerber** [comment]
@ajfisher I'm hoping with this audience that people will already have at least node installed and working which should save some time.  I think that encouraging people to work in pairs would be a good way to get around the problem too - if we can't get your environment set up after 30 minutes, team up with someone who has a working environment.  Good idea on pre-building the simple bot kits to save time.

Actually I think 90 minutes is a good time limit - enough to get set up, get some basic things happening (e.g. blink an LED, sweep a servo) and then with the exercises and nbd examples available online, it can be self-directed and people can work through it at their own pace. I'm happy for people to borrow the ardx kits outside of the workshop to work on their bots, so they'll be able to go see other talks and sessions rather than having to commit to a NodeBots workshop for 3 hours (if we did a double session). And then hopefully we'll get to see what they built in the lightning talks at the end of CampJS.

**Comment by timoxley** [comment]
What else needs to happen to make this a thing? Do you require a budget?

**Comment by wolfeidau** [comment]
I would be more than happy to volunteer as a helper and bring any stuff I have.

Really enjoyed the Melbourne one so keen to see this happen if possible.

**Comment by AnnaGerber** [comment]
I don't think we _need_ a budget because between all of us we probably have enough gear to run a basic workshop.

But if there was some budget for some extra arduinos, sensors etc for people to experiment with over the duration of campjs (and that we could keep as a pool of kit for future events), that would be great.

**Comment by timoxley** [comment]
@AnnaGerber interesting, how much is "some budget"?

**Comment by AnnaGerber** [comment]
@timoxley That big box of sensors I had with me last campjs sells for $50. Arduinos are around $30 each locally, or cheaper if we order them from China. The complete simplebot kits for node bots days ended up being about $80 each. So I'd say a minimum of $100, but the more budget we have, the more parts we could add.
